assets/js/admin/woocommerce_admin.js

```diff
@@ -49,10 +49,20 @@ jQuery( function ( $ ) {
 			}
 		})
 
-		.on( 'keyup', '.wc_input_price[type=text], .wc_input_decimal[type=text], .wc_input_country_iso[type=text], .wc-order-totals #refund_amount[type=text]', function() {
+		.on( 'change', '.wc_input_strict_price[type=text]', function() {
+			var regex    = new RegExp( '[^0-9\\' + woocommerce_admin.mon_decimal_point + ']+', 'gi' );
+			var value    = $( this ).val();
+			var newvalue = value.replace( regex, '' );
+
+			if ( value !== newvalue ) {
+				$( this ).val( newvalue );
+			}
+		})
+
+		.on( 'keyup', '.wc_input_price[type=text], .wc_input_decimal[type=text], .wc_input_country_iso[type=text]', function() {
 			var regex, error;
 
-			if ( $( this ).is( '.wc_input_price' ) || $( this ).is( '#refund_amount' ) ) {
+			if ( $( this ).is( '.wc_input_price' ) ) {
```

Comment:
Since this is being refactored, I'd be using `hasClass()` instead of `is()`. `is()` is slower and isn't necessary for checks of this kind.